### PR TITLE
feat: support `@cloudflare/workers-types` v4

### PR DESCRIPTION
**Pull request by lkwr**
Update the Cloudflare Workers types to the latest major.

This solve dependency installing warnings when using `@cloudflare/workers-types@^4` with remix.
There was no peer dependency for version 4 in the cloudflare packages.

### CODE REVIEW

File: .changeset/cuddly-cameras-collect.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@remix-run/cloudflare": patch
+"@remix-run/cloudflare-pages": patch
+"@remix-run/cloudflare-workers": patch
```
Comment (MichaelDeBoey) [inline .changeset/cuddly-cameras-collect.md:4]:
```suggestion
"@remix-run/cloudflare": minor
"@remix-run/cloudflare-pages": minor
"@remix-run/cloudflare-workers": minor
```

File: .changeset/cuddly-cameras-collect.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+---
+
+chore: update `@cloudflare/workers-types` to v4
```
Comment (MichaelDeBoey) [inline .changeset/cuddly-cameras-collect.md:7]:
```suggestion
feat: support `@cloudflare/workers-types` v4
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Comment by zhuhaow** [comment]
If possible, we should switch to v4 new importable types https://github.com/cloudflare/workerd/tree/main/npm/workers-types#importable-types and drop support of v2/v3.

The current way of referencing `@cloudflare/workers-types` doesn't work at all. Just create a new remix project with `create-remix`, run `pnpm typecheck` and it fails.

So this dependency version bump is probably not a change but a fix.

**Comment by MichaelDeBoey** [comment]
As this is a breaking change, we should do that in the next major version

I created #5732, which targets `v2` branch

**Comment by MichaelDeBoey** [comment]
Implemented by #5732

**Pull request closed without merging**